hibernate-core/src/main/java/org/hibernate/persister/entity/mutation/UpdateCoordinatorStandard.java

```diff
@@ -866,6 +889,84 @@ private void decomposeAttributeForUpdate(
 		}
 	}
 
+	private void decomposeEmbeddedAttributeMapping(
+			Object value,
+			SharedSessionContractImplementor session,
+			JdbcValueBindings jdbcValueBindings,
+			EntityTableMapping tableMapping,
+			boolean isDynamicUpdate,
+			boolean isDirty,
+			AttributeMapping attributeMapping) {
+		final CompositeGenerator compositeGenerator = (CompositeGenerator) attributeMapping.getGenerator();
+		attributeMapping.decompose(
+				value,
+				0,
+				jdbcValueBindings,
+				tableMapping,
+				(valueIndex, bindings, table, jdbcValue, jdbcMapping) -> {
+					if ( jdbcMapping instanceof EmbeddedAttributeMapping ) {
```

Comment:
Since `EmbeddedAttributeMapping` does not extend `SelectableMapping`, this branch does not make much sense, unless I am missing something.